uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-available-balance.component.js

```diff
@@ -37,11 +36,16 @@ define([
 
     /**
      * Implements the $onChanges controller method. It watches for changes in
-     * the component bindings.
+     * the component bindings. After bindings are ready, it loads dependencies,
+     * maps absence types with their entitlements, and then filters only absence
+     * types the contact is entitled to.
      */
     function $onChanges () {
       if (areBindingsReady()) {
-        loadDependencies();
+        loadDependencies().then(function () {
+          mapAbsenceTypesWithTheirEntitlements();
+          filterEntitledAbsenceTypes();
```

Comment:
please do not make up terms. there is no such a thing as an "entitled" absence type, so let's not call anything like that.